WarOOP/Models/Army.cs

```diff
@@ -26,6 +26,14 @@ public void AddUnits<T>(int count) where T : Warrior, new()
         }
     }
 
+    public void SetUnitsBehind()
```

Comment:
At first, if you want to use this approach, you need to make this initialization a part of object instantiation process. Because if the class which wants to use your army doesn't explicitly call this method, all your logic related to Lancer will fail.
The reason, I don't like this approach is that I don't think that the Warrior should know who stands behind him. It's not his responsibility to manage other warriors. It's a responsibility of army to know which warriors it contains and their state and the order.